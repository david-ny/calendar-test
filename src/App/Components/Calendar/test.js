import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Calendar from './index';
import { configTestData } from '../../test/testData';

const props = configTestData;

describe('Calendar', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Calendar {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Calendar {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
