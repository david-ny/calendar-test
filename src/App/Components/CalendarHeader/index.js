import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const CalendarHeader = ({ locale }) => {
  moment.locale(locale);
  return (
    <div
      className="calendar-header"
      style={{ display: 'flex' }}
    >
      {moment.weekdaysMin(true).map((item) => {
        return (
          <div
            key={item}
            style={{
             flexBasis: '14.2857%',
             maxWidth: '14.2857%',
             overflow: 'hidden',
            }}
          >
            {item}
          </div>
        );
      })}
    </div>
  );
};

CalendarHeader.propTypes = {
  locale: PropTypes.string,
};
CalendarHeader.defaultProps = {
  locale: 'en',
};

export default CalendarHeader;
