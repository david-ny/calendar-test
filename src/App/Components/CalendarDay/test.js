import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CalendarDay from './index';
import { CalendarDayTestData } from '../../test/testData';

const props = CalendarDayTestData;

describe('CalendarDay', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CalendarDay {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <CalendarDay {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
