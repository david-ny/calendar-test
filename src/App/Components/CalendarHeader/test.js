import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CalendarHeader from './index';
import { configTestData } from '../../test/testData';

const props = configTestData;

describe('CalendarHeader', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CalendarHeader {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <CalendarHeader {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
