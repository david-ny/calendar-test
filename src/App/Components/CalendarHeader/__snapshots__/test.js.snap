// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader has a valid snapshot 1`] = `
<div
  className="calendar-header"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "flexBasis": "14.2857%",
        "maxWidth": "14.2857%",
        "overflow": "hidden",
      }
    }
  >
    Su
  </div>
  <div
    style={
      Object {
        "flexBasis": "14.2857%",
        "maxWidth": "14.2857%",
        "overflow": "hidden",
      }
    }
  >
    Mo
  </div>
  <div
    style={
      Object {
        "flexBasis": "14.2857%",
        "maxWidth": "14.2857%",
        "overflow": "hidden",
      }
    }
  >
    Tu
  </div>
  <div
    style={
      Object {
        "flexBasis": "14.2857%",
        "maxWidth": "14.2857%",
        "overflow": "hidden",
      }
    }
  >
    We
  </div>
  <div
    style={
      Object {
        "flexBasis": "14.2857%",
        "maxWidth": "14.2857%",
        "overflow": "hidden",
      }
    }
  >
    Th
  </div>
  <div
    style={
      Object {
        "flexBasis": "14.2857%",
        "maxWidth": "14.2857%",
        "overflow": "hidden",
      }
    }
  >
    Fr
  </div>
  <div
    style={
      Object {
        "flexBasis": "14.2857%",
        "maxWidth": "14.2857%",
        "overflow": "hidden",
      }
    }
  >
    Sa
  </div>
</div>
`;
